lib/laa_crime_forms_common/pricing/nsm/calculators/additional_fees.rb

```diff
@@ -0,0 +1,54 @@
+module LaaCrimeFormsCommon
+  module Pricing
+    module Nsm
+      module Calculators
+        class AdditionalFees
+          ELLIGIBLE_DATE = Date.new(2024, 12, 7)
+          class << self
+            def call(claim, rates: Rates.call(claim))
+              new(claim, rates).call
+            end
+          end
+
+          def initialize(claim, rates)
+            @claim = claim
+            @rates = rates
+          end
+
+          def call
+            {
+              youth_court_fee:,
+            }
+          end
+
+        private
+
+          def claim_date
+            case claim.claim_type
+            when "non_standard_magistrate"
+              claim.rep_order_date
+            when "breach_of_injunction"
+              claim.cntp_date
+            else
+              raise "Unrecognised claim type '#{claim.claim_type}'"
+            end
+          end
+
+          def youth_court_fee
+            claim.youth_court_fee_claimed && youth_court_fee_applicable ? rates.additional_fees[:youth_court] : BigDecimal("0")
+          end
+
+          def youth_court_fee_applicable
+            [
+              claim_date >= ELLIGIBLE_DATE,
+              claim.youth_court == "yes",
+              %w[category_1a category_2a].include(claim.plea_category),
```

Comment:
`claim.plea_category.match?(/category_[12]a/)`

Reply:
leaving this as it is for now - purely because I feel it's more readable (this might be a personal take because I hate regex) and I want to reduce the cognitive load of reading the code. That said, if the criteria later down the line ends up including more categories e.g. (category 3a, category 4a...so on and so forth) - we can move to regex, as having a list could get unruly

Reply:
(ofc this is now referring to the youth_court_fee calculator as I renamed it)